extensions/machinepools/digitalocean_machine_config.go

```diff
@@ -48,7 +49,7 @@ func NewDigitalOceanMachineConfig(generatedPoolName, namespace string) []unstruc
 		machineConfig.SetGenerateName(generatedPoolName)
 		machineConfig.SetNamespace(namespace)
 
-		machineConfig.Object["accessToken"] = ""
+		machineConfig.Object["accessToken"] = doMachineConfig.AccessToken
```

Comment:
can you double check this, I'm pretty sure it is supposed to be blank for the machine / node templates. Otherwise, you'd need to update the other files in this folder that set it to blank for say linode.